alacritty/src/display/window.rs

```diff
@@ -471,7 +471,24 @@ impl Window {
         self.windowed_context.swap_buffers_with_damage(damage).expect("swap buffes with damage");
     }
 
+    #[cfg(any(target_os = "macos", windows))]
     pub fn swap_buffers_with_damage_supported(&self) -> bool {
+        // Disable damage tracking on macOS/Windows since there's no observation of it working
+        false
+    }
+
+    #[cfg(not(any(target_os = "macos", windows)))]
+    pub fn swap_buffers_with_damage_supported(&self) -> bool {
+        // On X11 damage tracking is behaving in unexpected way on some NVIDIA systems. Given that
+        // there're no compositors supporting damage tracking and there's no such thing on GLX,
+        // which is used by default, disabling it.
+        //
+        // For more see https://github.com/alacritty/alacritty/issues/6051.
```

Comment:
```suggestion
        // On X11 damage tracking is behaving in unexpected ways on some NVIDIA systems. Since there's no compositor supporting it, damage tracking is disabled on X11.
        //
        // For more see https://github.com/alacritty/alacritty/issues/6051.
```

I think we can omit the GLX part, since the relevant portion would still be the compositor support. That way it reads a little nicer.